lib/pharos/ssh/client.rb

```diff
@@ -48,23 +59,24 @@ def bastion
       # @param options [Hash] see Net::SSH#start
       def connect(**options)
         synchronize do
-          logger.debug { "connect: #{@user}@#{@host} (#{@opts})" }
-          if bastion
-            gw_opts = {}
-            gw_opts[:keys] = [bastion.ssh_key_path] if bastion.ssh_key_path
-            gateway = Net::SSH::Gateway.new(bastion.address, bastion.user, gw_opts)
```

Comment:
Becuse `ssh.gateway` returned `ssh.gateway.open`, the `Net::SSH::Gateway.new` was duplicated here and was left in a "dangling" local variable that could not be accessed for `shutdown!`.